searx/webapp.py

```diff
@@ -683,9 +693,12 @@ def search():
 
     # output_format
     output_format = request.form.get('format', 'html')
-    if output_format not in ['html', 'csv', 'json', 'rss']:
+    if output_format not in OUTPUT_FORMATS:
         output_format = 'html'
 
+    if output_format not in get_value(settings, 'search', 'formats', default=OUTPUT_FORMATS):
+        flask.abort(403)
```

Comment:
Looking at the documentation: https://flask.palletsprojects.com/en/2.0.x/api/#flask.abort

There is no need for an exception, just `return 'Access denied', 403` is enough.

Reply:
Hm, there is a difference .. 

AFAIK:  just a `return 'Access denied', 403` will be handled by the flask app (Flask generates a 403 response with the given string).   `flask.abort(403)` will raise an exception and the 403 page comes from the HTTP server.

If I was unclear, try both versions in a local instance (make run).

The question for me is: what do we want, should flask build the response or should the 403 be propagated over the http server? I decided to give the 403 back to the http server.

Reply:
> the 403 page comes from the HTTP server.

So either Werkzeug or uwsgi ?

>  I decided to give the 403 back to the http server.

why ?

Reply:
I don't know if it helps, but Flask's documentation says: [Redirects and Errors: ... to abort a request early with an error code, use the abort() function:](https://flask.palletsprojects.com/en/2.0.x/quickstart/#redirects-and-errors)

Reply:
> AFAIK: Whatever (error page) is configured in the http server. Does not depend on the technology which is used to bind the Flask-App.

I wish to reduce the number of configuration locations.

But it returns an HTTP 403 whatever the solution.